### DISCUSSION TITLE
Any way to debug failed vercel.json validations?

### DISCUSSION THREAD

**Discussion by selalipop** [vercel/community#3147] [Help] [answered]
### Summary

I had a deploy failing with no logs. The docs indicate that an improperly formatted vercel.json can cause this.

I chased it down to being a limitation on the max execution duration for the hobby plan, but it wasn't a great first experience on the platform.

Is there a better way to debug issues with the config defined in `vercel.json`? Specific issues have been raised about failed configs, but I'm looking for a general approach to finding what exactly went wrong for a given config file

### Steps to Reproduce

Create a configuration with an invalid value, like 60 seconds on a hobby account

```
{
  "functions": {
    "src": {"maxDuration": 60}
  }
}
```

**Comment by amyegan** [comment, marked as answer]
Hi @selalipop. You will see an error message printed about it in your PR if you have one opened or in the terminal if you use [Vercel CLI](https://vercel.com/docs/cli). 

Maximum execution duration is currently 10 seconds for Hobby and 60 seconds for Pro. So the warning I see when trying to use that config on a Hobby project is `The value for maxDuration must be between 1 second and 10 seconds, in order to increase this limit upgrade your plan: https://vercel.com/pricing`

You can find more info about configuring `functions` in `vercel.json` at https://vercel.com/docs/concepts/projects/project-configuration#functions. And more info on duration limits can be found at https://vercel.com/docs/concepts/limits/overview#serverless-function-execution-timeout

**Reply by asindhu** [reply]
Hi, I have the same issue related to adding a cron job. Vercel doesn't pick up the commit with the `vercel.json` file at all, so I see nothing there, just an error in GitHub that says "Vercel - Deployment failed". I'm basically trying to do exactly the same as [this one](https://github.com/orgs/vercel/discussions/2315) except I'm on production so it's not an issue of it being a preview branch. Same exact `vercel.json` file as in that example. Could you provide a bit more detail on how to debug? Don't know what PR means or how to use the Vercel CLI, would really appreciate a more detailed step-by-step on how to get the relevant error message, thank you.

**Reply by amyegan** [reply]
Hey @asindhu. These are the steps I recommend for local debugging: 
1. [Install Vercel CLI](https://vercel.com/docs/cli#installing-vercel-cli) with command `npm i -g vercel`
2. Use [`vercel login`](https://vercel.com/docs/cli/login) to connect with your Vercel account
3. Run [`vercel build --prod`](https://vercel.com/docs/cli/build) in the local copy of your repository. This should be same folder where you would run commands like `npm run dev`. There will be a prompt to connect with an existing project or create a new one, and possibly a few more questions dependent upon your answers.
4. Check the build output for error and warning details.

**Reply by asindhu** [reply]
Awesome thank you for the quick reply! Finally found the issue -- you had actually said it in a previous answer but I missed it... I wanted to test the cron job so I set it to every minute but it seems that hobby projects are limited to once a day. That being said, the error actually never showed up anywhere, I just found out by playing around with different cron settings. I followed your instructions and ran the build through the Vercel CLI, but it showed no issues even with the improperly configured cron job. I guess it's an edge case in terms of functionality being limited by the hobby tier but it would be great to have some kind of visible error for this :) Thanks again for the help.
